### ISSUE TITLE
Cannot pip install pymatbridge

### ISSUE THREAD

**Issue by brunobeltran** [arokem/python-matlab-bridge#214]
Title says it all. Due to the fact that the setup function requires the get_messenger_dir function, it imports one of pymatbridge's internal modules, which causes an unecessary import of nmpy and pyzmq before pip's install_requires can pull them in.

This can be solved in several ways. The simplest might be to separate this function into its own module that is imported by both setup.py and pymatbridge.messenger.make.

@arokem I'm not sure how strongly you feel about the current organization of the project. If this simple change makes sense to you I can make another pull request.

**Comment by arokem** [comment]
Sorry for the delay in response. I just verified that I do see the problem you mentioned. Yes - please do make a PR with the suggested change - it sounds like a good direction to me.

**Comment by brunobeltran** [comment]
Some of my thoughts on the implementation on what I ended up doing. 

It looks like make.py in the messenger directory is already a nicely separated spot for "building only" functions.

As such, I went with the same solution as the Numpy developers, and simply changed __init__.py to detect if we're running setup.py and only load those functions in that case. 

Thanks to this, when running setup.py, we no longer need any of our dependencies.

Also, updated extras_require (wasn't being used before) to the "correct" packages and some guaranteed good version numbers.
